packages/core/scripts/collateral-umip/1_Propose.js

```diff
@@ -46,31 +58,50 @@ async function runExport() {
 
   const collaterals = _.castArray(argv.collateral);
   const fees = _.castArray(argv.fee);
-  const decimals = argv.decimals && _.castArray(argv.decimals);
-
-  if (collaterals.length !== fees.length || (decimals && decimals.length !== collaterals.length)) {
-    throw new Error("Must provide the same number of elements to --fee, --collateral, and --decimals (optional)");
+  const decimals = _.castArray(argv.decimals);
+  const polygonCollaterals = _.castArray(argv.polygonCollateral);
+
+  if (
+    collaterals.length !== fees.length ||
+    (decimals && decimals.length !== collaterals.length) ||
+    (polygonCollaterals && polygonCollaterals.length !== collaterals.length)
+  ) {
+    throw new Error(
+      "Must provide the same number of elements to --fee, --collateral, and optional flags --decimals and --polygonCollateral"
+    );
   }
 
-  const argObjects = _.zipWith(collaterals, fees, decimals, (collateral, fee, numDecimalsArg) => {
-    return { collateral, fee, numDecimalsArg };
-  });
+  const argObjects = _.zipWith(
+    collaterals,
+    fees,
+    decimals,
+    polygonCollaterals,
+    (collateral, fee, numDecimalsArg, polygonCollateral) => {
+      return { collateral, fee, numDecimalsArg, polygonCollateral };
+    }
+  );
 
-  const getTxns = async ({ collateral, fee, numDecimalsArg }) => {
+  const getTxns = async ({ collateral, fee, numDecimalsArg, polygonCollateral }) => {
     const decimals = await getDecimals(collateral, numDecimalsArg, ERC20);
     console.log("Examining collateral", collateral);
     const convertedFeeAmount = parseUnits(fee, decimals).toString();
     console.log(`Fee in token's decimals: ${convertedFeeAmount}`);
 
     const txns = [];
 
-    // The proposal will first add a final fee for the currency.
+    // The proposal will first add a final fee for the currency if the current final fee is different from the
+    // proposed new one.
     const store = await Store.deployed();
-    const addFinalFeeToStoreTx = store.contract.methods
-      .setFinalFee(collateral, { rawValue: convertedFeeAmount })
-      .encodeABI();
-    console.log("addFinalFeeToStoreTx", addFinalFeeToStoreTx);
-    txns.push({ to: store.address, value: 0, data: addFinalFeeToStoreTx });
+    const currentFinalFee = await store.computeFinalFee(collateral);
+    if (currentFinalFee.toString() !== convertedFeeAmount) {
```

Comment:
There are now situations where the Ethereum Store shouldn't be modified, but the Polygon Store needs to update final fees, so we should only update the Ethereum store when final fees change.